src/AccessLogs.sol

```diff
@@ -0,0 +1,190 @@
+// Copyright 2023 Cartesi Pte. Ltd.
+
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+/// @dev This file is generated from helper_scripts/generate_AccessLogs.sh, one should not modify the content directly
+
+pragma solidity ^0.8.0;
+
+import "./Memory.sol";
+import "./UArchConstants.sol";
+
+library AccessLogs {
+    using AccessLogs for bytes;
+    using Memory for Memory.AlignedSize;
+
+    struct Context {
+        bytes32 currentRootHash;
+        bytes buffer;
+        uint128 pointer;
+    }
```

Comment:
I'm thinking, should we make the pair buffer and pointer their own type? Something like:

```
struct Buffer {
    bytes data;
    uint64 pointer;
}

struct Context {
    bytes32 currentRootHash;
    Buffer buffer;
}
```

I think it would make things easier. Could even be on its own library.

Also, any special reason to user a `uint128`?

Reply:
The pointer was type `uint128` in the first version you wrote. I didn't change it. Are you suggesting to use `uint64` or `uint256`?

Reply:
I think either one is fine. A `uint64` is enough, but I think it won't occupy less space than a `uint256` because of padding. And I think a `uint256` uses less gas because it's the size of a word in the EVM.

So I think we should use `uint256`, and if we add more stuff to the struct later that would make a `uint64` take less space, we should consider changing.

What do you think?